syntax = "proto3";

package db;

message SdvxSong {
    int32 id = 1;
    string title = 2;
    string nick1 = 3;
    string nick2 = 4;
    string nick3 = 5;

    int32 level = 6;
    string code = 7;
    string bpm = 8;

    string artist = 9;
    string effector = 10;
    string illustrator = 11;

    int32 chain_v = 12;
    int32 chain_vi = 13;

    string table_S = 14;
    string table_PUC = 15;
}

message SdvxList{
    repeated SdvxSong sdvxsongs = 1;
}

message Coord {
    int32 x = 1;
    int32 y = 2;
}

message TableList {
    map<string, Coord> dict = 1;
}

message SingleSearch {
    SdvxSong song = 1;
    int32 lcs = 2;
}

message SearchResult {
    repeated SingleSearch result = 1;
}